nebari/template/stages/07-kubernetes-services/modules/kubernetes/services/argo-workflows/main.tf

```diff
@@ -232,3 +266,230 @@ resource "kubernetes_cluster_role_binding" "argo-view-rb" {
+              "workflows",
+            ]
+          },
+        ]
+        "sideEffects" = "None"
+      },
+    ]
+  }
+}
+
+resource "kubernetes_manifest" "validatingwebhookconfiguration_admission_controller" {
+  manifest = {
+    "apiVersion" = "admissionregistration.k8s.io/v1"
+    "kind"       = "ValidatingWebhookConfiguration"
+    "metadata" = {
+      "name" = "wf-admission-controller"
+    }
+    "webhooks" = [
+      {
+        "admissionReviewVersions" = [
+          "v1",
+          "v1beta1",
+        ]
+        "clientConfig" = {
+          "url" = "https://${var.external-url}/${local.argo-workflows-prefix}/validate"
+        }
+        "name" = "wf-validating-admission-controller.${var.namespace}.svc"
+        "rules" = [
+          {
+            "apiGroups" = [
+              "argoproj.io",
+            ]
+            "apiVersions" = [
+              "v1alpha1",
+            ]
+            "operations" = [
+              "CREATE",
+            ]
+            "resources" = [
+              "workflows",
+            ]
+          },
+        ]
+        "sideEffects" = "None"
+      },
+    ]
+  }
+}
+
+resource "kubernetes_manifest" "deployment_admission_controller" {
+  manifest = {
+    "apiVersion" = "apps/v1"
+    "kind"       = "Deployment"
+    "metadata" = {
+      "name"      = "nebari-workflow-controller"
+      "namespace" = var.namespace
+    }
+    "spec" = {
+      "replicas" = 1
+      "selector" = {
+        "matchLabels" = {
+          "app" = "nebari-workflow-controller"
+        }
+      }
+      "template" = {
+        "metadata" = {
+          "labels" = {
+            "app" = "nebari-workflow-controller"
+          }
+        }
+        "spec" = {
+          serviceAccountName           = kubernetes_service_account.wf-admission-controller.metadata.0.name
+          automountServiceAccountToken = true
+          "containers" = [
+            {
+              command = ["sleep", "1000000"]
+              "env" = [
+                {
+                  "name" = "KEYCLOAK_USERNAME"
+                  "valueFrom" = {
+                    "secretKeyRef" = {
+                      "key"  = "username"
+                      "name" = "keycloak-read-only-user-credentials"
+                    }
+                  }
+                },
+                {
+                  "name" = "KEYCLOAK_PASSWORD"
+                  "valueFrom" = {
+                    "secretKeyRef" = {
+                      "key"  = "password"
+                      "name" = "keycloak-read-only-user-credentials"
+                    }
+                  }
+                },
+                {
+                  "name"  = "KEYCLOAK_URL"
+                  "value" = "https://${var.external-url}/auth/"
+                },
+                {
+                  "name"  = "NAMESPACE"
+                  "value" = var.namespace
+                },
+              ]
+              # "image" = "quay.io/nebari/nebari-workflow-controller:main"
```

Comment:
It might be worth adding a `WORKFLOW_CONTROLLER_IMAGE_TAG` to the constants.py so we can update it in one location. On second thought, using the `nebari_version` might be a little tricky when we're developing since there might not exist an image for the dev version we're working on.

Reply:
Nice, I went ahead and added this.